### PR TITLE
Test seq io unittest

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request replaces the print-and-compare test in test_SeqIO.py by a unittest.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Tests/test_SeqIO.py [hunk 1]
```diff
@@ -369,357 +125,3284 @@ def alignment_summary(alignment, index=" "):
...
+        # Some alignment file formats have magic characters which mean
+        # use the letter in this position in the first sequence.
+        # They should all have been converted by the parser, but if
+        # not reversing the record order might expose an error.  Maybe.
+        records.reverse()
+        self.check_simple_write_read(records, t_format, t_count,
+                                     expected_messages)
+
+    def test_sff1(self):
+        records = [SeqRecord(Seq("tcagGGTCTACATGTTGGTTAACCCGTACTGATTTGAATT...GGGCTTa"), "E3MFGYR02JWQ7T", "E3MFGYR02JWQ7T"),
```
Comment (peterjc) [inline Tests/test_SeqIO.py]: I find it rather misleading that you make a list of expected ``SeqRecord`` objects but with truncated sequences. I would find it clearer to have a list of expected identifiers (strings in full) and truncated sequences - to match the list of expected sequence lengths.

**Comment by peterjc** [comment]
It will be interesting to see if the automated test coverage numbers are informative here...

File: Tests/test_SeqIO.py (hunk 1 above)
Comment (peterjc) [reply Tests/test_SeqIO.py]: Update looks clearer to me now, thanks!

Comment (mdehoon) [reply Tests/test_SeqIO.py]: OK to proceed? The remaining AppVeyor failure is unrelated to `test_SeqIO.py`.

**Comment by peterjc** [comment]
Agreed, thank you.

**Pull request merged**